cmd/sovereignnode/sovereignNodeRunner.go

```diff
@@ -1664,11 +1652,17 @@ func (snr *sovereignNodeRunner) CreateArgsRunTypeComponents(coreComp mainFactory
 		return nil, fmt.Errorf("NewRunTypeComponentsFactory failed: %w", err)
 	}
 
+	sovHeaderSigVerifier, err := headerCheck.NewSovereignHeaderSigVerifier(cryptoComp.BlockSigner())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is **NOT OK**.

Why would you create it outside runtype and then pass it on args? you need to create it inside `sovRunTypeComp`.
For this you probably need to pass crypto components to run type sov

Reply:
To minimize the conflicts with the next PRs, I created another PR for this at the end of refactor 
https://github.com/multiversx/mx-chain-go/pull/6140